### PR TITLE
Fix revisions

### PR DESCRIPTION
**Pull request by mattheu**
I'm getting an error when saving posts caused by this line.

Basically, we can't do it like this. If we want to do revisions of the data, we should require the use of this meta revisions plugin. https://wordpress.org/plugins/wp-post-meta-revisions/.

This adds the approprate filters to revision the data, and add some instructions to the readme.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -88,12 +92,12 @@ array(
...
-array( 
-	'name'  => 'image', 
-	'label' => 'Test Image', 
-	'type'  => 'attachment', 
-	'config' => array( 
-		'button_text' => 'Custom Button Text' 
-	) 
+array(
+	'name'  => 'image',
+	'label' => 'Test Image',
```
Comment (BronsonQuick) [inline README.md]: This should be translatable.

File: README.md [hunk 2]
```diff
@@ -88,12 +92,12 @@ array(
...
+	'type'  => 'attachment',
+	'config' => array(
+		'button_text' => 'Custom Button Text'
```
Comment (BronsonQuick) [inline README.md]: This should be translatable too.

**Comment by mattheu** [comment]
Thanks @BronsonQuick fixed translations

**Comment by mattheu** [comment]
For reference. The problem is this section in `wp_save_post_revision`

``` php
foreach ( array_keys( _wp_post_revision_fields( $post ) ) as $field ) {
    if ( normalize_whitespace( $post->$field ) != normalize_whitespace( $last_revision->$field ) ) {
        $post_has_changed = true;
        break;
    }
}
```

You must pass `normalize_whitespace` a string, and not an array. Which is what our post meta fields contain.

**Pull request merged**